Marlin/src/module/stepper.cpp

```diff
@@ -2843,7 +2870,39 @@ hal_timer_t Stepper::block_phase_isr() {
         calc_nonlinear_e(current_block->initial_rate << oversampling_factor);
       #endif
 
-      #if ENABLED(LIN_ADVANCE)
+      #if ENABLED(LA_ZERO_SLOWDOWN)
+        /**
+         * 1. [x] compensate for jerk and line width changes
+         * 2. [x] keep computations in movement step scale
+         * 3. [ ] precompute a_max in movement step scale and inter block correction factors in the planner
+         * 4. [ ] use int arithmetic
+         */
+        if (current_block->step_event_count != 0) {
+          curr_step_rate = current_block->initial_rate;
+          const float old_xy_to_e_steps = xy_to_e_steps;
+          if (current_block->steps.e) {
+            xy_to_e_steps = float(current_block->steps.e) / float(current_block->step_event_count);
```

Comment:
This var is called `xy_to_e_steps`, but technically computes the inverse. Not critical, but I think you can actually save one division, if you define it as it's named. Go
```c++
          xy_to_e_steps = float(current_block->step_event_count) / float(current_block->steps.e);
```
here and follow up with
```c++
          const float xy_to_e_steps_ratio = xy_to_e_steps / old_xy_to_e_steps;
          current_la_step_count = current_la_step_count * xy_to_e_steps_ratio;
          current_la_step_rate = current_la_step_rate * xy_to_e_steps_ratio;
          a_max = float(planner.max_acceleration_steps_per_s2[E_AXIS + E_INDEX_N(extruder)]) * xy_to_e_steps;
```
at lines 2896 - 2899.

Reply:
My intent was that that's the factor to convert xy steps to e steps:
`xy_to_e = total_e/total_xy`

So then I can multiply xy steps by it to get e steps.

Regardless, I'll take a look at storing the inverse as you say to do less divisions.